apps/ops-func/src/adapters/cosmos/message-metadata-deleter.ts

```diff
@@ -0,0 +1,50 @@
+import { Container } from "@azure/cosmos";
+
+interface DeleteMessageMetadataResponse {
+  readonly statusCode?: number;
+  readonly success: boolean;
+}
+
+export interface MessageMetadataDeleter {
+  deleteMessageMetadata: (
+    partitionKey: string,
+    id: string,
+  ) => Promise<DeleteMessageMetadataResponse>;
+}
+
+export class CosmosMessageMetadataDeleter implements MessageMetadataDeleter {
+  container: Container;
+
+  constructor(container: Container) {
+    this.container = container;
+  }
+
+  /**
+   * Deletes a message metadata from the container if it exists.
+   *
+   * @param fiscalCode - The partition key of the item.
+   * @param messageId - The id of the item.
+   * @returns A promise that resolves when the deletion is complete.
+   */
+  async deleteMessageMetadata(
+    fiscalCode: string,
+    messageId: string,
```

Comment:
Can't we use branded types? we should have a `FiscalCode` type, maybe we need to add the `MessageId` type as well

Reply:
I was thinking that a potential use case for this function could be handling messages with malformed messageId or FiscalCode, adding this strict check at this stage might prevent us from deleting it later. What's your take?

Reply:
good point 👍 please explain this rationale in a comment